opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -186,72 +235,86 @@ def __init__(
             unit,
             value_type,
+            meter,
             label_keys=label_keys,
             enabled=enabled,
             monotonic=monotonic,
+            absolute=absolute,
         )
 
-    def set(self, label_set: LabelSet, value: metrics_api.ValueT) -> None:
+    def set(self, value: metrics_api.ValueT, label_set: LabelSet) -> None:
         """See `opentelemetry.metrics.Gauge.set`."""
         self.get_handle(label_set).set(value)
 
     UPDATE_FUNCTION = set
 
 
 class Measure(Metric, metrics_api.Measure):
-    """See `opentelemetry.metrics.Measure`.
-
-    By default, measure metrics can accept both positive and negatives.
-    Negative inputs will be discarded when monotonic is True.
-    """
+    """See `opentelemetry.metrics.Measure`."""
 
     HANDLE_TYPE = MeasureHandle
 
-    def __init__(
-        self,
-        name: str,
-        description: str,
-        unit: str,
-        value_type: Type[metrics_api.ValueT],
-        label_keys: Sequence[str] = (),
-        enabled: bool = False,
-        monotonic: bool = False,
-    ):
-        super().__init__(
-            name,
-            description,
-            unit,
-            value_type,
-            label_keys=label_keys,
-            enabled=enabled,
-            monotonic=monotonic,
-        )
-
-    def record(self, label_set: LabelSet, value: metrics_api.ValueT) -> None:
+    def record(self, value: metrics_api.ValueT, label_set: LabelSet) -> None:
         """See `opentelemetry.metrics.Measure.record`."""
         self.get_handle(label_set).record(value)
 
     UPDATE_FUNCTION = record
 
 
+class Record:
+    """Container class used for processing in the `Batcher`"""
+
+    def __init__(
+        self, metric: Metric, label_set: LabelSet, aggregator: Aggregator
+    ):
+        self.metric = metric
+        self.label_set = label_set
+        self.aggregator = aggregator
+
+
 # Used when getting a LabelSet with no key/values
 EMPTY_LABEL_SET = LabelSet()
 
 
 class Meter(metrics_api.Meter):
-    """See `opentelemetry.metrics.Meter`."""
+    """See `opentelemetry.metrics.Meter`.
+
+    Args:
+        batcher: The `Batcher` used for this meter.
+    """
 
-    def __init__(self):
-        self.labels = {}
+    def __init__(self, batcher: Batcher = UngroupedBatcher(True)):
+        self.batcher = batcher
+        # label_sets is a map of the labelset's encoded value to the label set
+        self.label_sets = {}
+        self.metrics = set()
+
+    def collect(self) -> None:
+        """Collects all the metrics created with this `Meter` for export.
+
+        Utilizes the batcher to create checkpoints of the current values in
+        each aggregator belonging to the metrics that were created with this
+        meter instance.
+        """
+        for metric in self.metrics:
+            if metric.enabled:
+                for label_set, handle in metric.handles.items():
+                    # Consider storing records in memory?
```

Comment:
Maybe adding TODO for this possible optimization, this comment could be easily lost